src/server/input/surface_input_dispatcher.cpp

```diff
@@ -150,7 +152,10 @@ mi::SurfaceInputDispatcher::SurfaceInputDispatcher(std::shared_ptr<mi::Scene> co
       started(false)
 {
     scene_observer = std::make_shared<InputDispatcherSceneObserver>(
-        [this](ms::Surface* s){surface_removed(s);},
+        std::bind(
+            std::mem_fn(&SurfaceInputDispatcher::surface_removed),
+            this,
+            std::placeholders::_1),
```

Comment:
Lambdas should be preferred to std::bind (that's so very C++11). Something like:
```c++
        [this](auto const& s) { surface_removed(s); },
```